theano/compile/function_module.py

```diff
@@ -752,9 +753,33 @@ def checkSV(sv_ori, sv_rpl):
         return f_cpy
 
     def __call__(self, *args, **kwargs):
+        """
+        Evaluates value of a function on given arguments.
+
+        Parameters
+        ----------
+        args : list
+            List of inputs to the function. All inputs are required, even when
+            some of them are not necessary to calculate requested subset of
+            outputs.
+
+        kwargs : dict
+            TODO: other kwargs?
+            Keyword argument `output_subset` is a list of indices of the
+            function's outputs that are requested to be calculated.
```

Comment:
It's worth noting that the issue arose at all, because to do 2) we would still have to pass the information about the output subset to the __call__ anyway, and would have to do it indirectly (both all regular and keyword arguments are already potentially being used).